src/simulator_core/adapter/transforms/string_to_esdl.py

```diff
@@ -8,15 +8,28 @@ class StringEsdlAssetMapper:
     Please note that the str_to_type_dict needs to have unique keys and values.
     """
 
+    type_to_str_dict = {
```

Comment:
Waarom heb je deze toegevoegd? Ik heb bewust maar 1 dict erin gezet omdat je anders twee dicts moet hebben. Met 1 dict en de oude zoek methode gerandeer je dat je altijd in dezelfde dict zoekt.

Reply:
Deze hebben we nodig omdat er dus meer producers zijn. In jouw voorbeeld netwerk zit een "GenericProducer" waardoor je dus niet kan verwijzen naar één key in een dict. 

Het kan natuurlijk ook zijn dat het voorbeeld/test niet goed is?

Reply:
Reformatted.